pkg/metrics/metrics.jsx

```diff
@@ -370,6 +391,81 @@ class CurrentMetrics extends React.Component {
                 });
     }
 
+    onPrivilegedMetricsUpdate(event, message) {
+        debug("process metrics message", message);
+        const data = JSON.parse(message);
+
+        if (!Array.isArray(data)) {
+            this.processNames = data.metrics[0].instances;
+            return;
+        }
+
+        data.forEach(privilegedSamples => decompress_samples(privilegedSamples, this.privilegedSamples));
+
+        // return [ name, read, write, isUser, isContainer, uid | cgroup ]
+        const n_biggest = (n, names, valuesA, valuesB) => {
+            const merged = [];
+            const userSlices = {};
+            names.forEach((name, i) => {
+                // filter out invalid values, the empty (root) cgroup, non-services
+                if (name.endsWith('.service')) {
+                    // only keep cgroup basenames, and drop redundant .service suffix
+                    const label = name.replace(/.*\//, '').replace(/\.service$/, '');
+                    const isUser = name.match(useridCgroupRe);
+                    merged.push([label, valuesA[i], valuesB[i], isUser, false, name]);
+                    return;
+                }
+
+                // filter out podman containers, but only for the logged in
+                // user or root user if the user is privileged. Other users
+                // containers will show up under the user@uid cgroup
+                const matches = name.match(/libpod-(?<containerid>[a-z|0-9]{64})\.scope/);
+                if (matches && valuesA[i] && valuesB[i]) {
+                    let isUser = false;
+                    let uid = 0;
+                    const containerid = matches.groups.containerid;
+                    const umatches = name.match(useridCgroupRe);
+                    if (umatches) {
+                        isUser = true;
+                        uid = parseInt(umatches.groups.userid);
+                    }
```

Comment:
These 9 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17692-20220912-104552-88be796a-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)